### ISSUE TITLE
CompositeFuture forces you to use raw types

### ISSUE THREAD

**Issue by ctron** [eclipse-vertx/vert.x#2058]
There are currently three methods in `CompositeFuture` which force you to use raw types:
* `io.vertx.core.CompositeFuture.all(List<Future>)`
* `io.vertx.core.CompositeFuture.any(List<Future>)`
* `io.vertx.core.CompositeFuture.join(List<Future>)`

The causes compiler warnings to appear all over the place:

![image](https://user-images.githubusercontent.com/202474/28312085-1fe7c068-6bb2-11e7-8b33-07c96e89fbe2.png)

It should be possible to rewrite the method signature to e.g. `io.vertx.core.CompositeFuture.all(List<Future<?>>)`, which solves the raw types warning, but is not compatible with the previous declaration and this would break compatibility.

However there could be be a set of workarounds. One could be simply adding new methods like e.g. `allOf`. Or putting everything in a new interface `TypedCompositeFuture`, which would avoid the collision.

**Comment by jan-zajic** [comment]
See Closed PR #1994 with discussion

**Comment by tmbull** [comment]
Per discussion on #3595, it sounds like this can be fixed now?

**Comment by vietj** [comment]
yes, we had an attempt but it creates breaking change, so the consensus
would be to copy the relevant methods to Future and deprecate the others

**Comment by lukasalexanderweber** [comment]
@vietj are there any updates on this? :)

**Comment by vietj** [comment]
that shall be a vertx 5 item

**Comment by lukasalexanderweber** [comment]
@vietj I just happened to see that this seems to be solved: https://github.com/vert-x3/vertx-web/commit/e299227694b48246beda152b1cfea60f129b9d7d
